File: offer-store-service/src/main/java/pl/allezon/offerstore/services/OfferEndpoint.java
package pl.allezon.offerstore.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import pl.allezon.offerstore.domain.Offer;
import pl.allezon.offerstore.repository.OfferRepository;

@RestController
@RequestMapping("/offers")
public class OfferEndpoint {

    private final OfferRepository offerRepository;

    @Autowired
    public OfferEndpoint(OfferRepository offerRepository) {
        this.offerRepository = offerRepository;
    }

    @GetMapping("/{id}")
    public Offer getOffer(@PathVariable Long id) {
        return offerRepository.findOne(id);
    }
}

File: offer-store-service/src/main/scala/pl/allezon/offerstore/service/OfferEndpoint.scala
package pl.allezon.offerstore.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.{GetMapping, PathVariable, RequestMapping, RestController}
import pl.allezon.offerstore.domain.Offer
import pl.allezon.offerstore.repository.OfferRepository

@RestController
@RequestMapping(Array("/offers"))
class OfferEndpoint @Autowired()(val offerRepository: OfferRepository) {

  @GetMapping(Array("/{id}"))
  def getOffer(@PathVariable id: String): Offer = offerRepository.findById(id)
}
